### ISSUE TITLE
GetByLabel and aria-labelledby

### ISSUE THREAD

**Issue by Clement-Hue** [callstack/react-native-testing-library#1563]
## Describe the bug

When I use aria-labelledby and getByLabel, the test doesn't work.
However I have no problem if I use aria-label 

## Expected behavior

The test should pass

## Steps to Reproduce
Here is the component:
```ts
const Switch: React.FC<SwitchProps> = ({checked, onPress, title, ...props}) => {
  const theme = useTheme();
  const id = useId();
  return (
    <View style={switchStyle.container}>
      <BaseSwitch
        thumbColor={
          checked ? theme.color.primary['50'] : theme.color.grey['50']
        }
        trackColor={{
          true: theme.color.primary['30'],
          false: theme.color.grey['30'],
        }}
        value={checked}
        onValueChange={onPress}
        aria-labelledby={id}
        {...props}
      />
      {!!title && (
        <Text nativeID={id} style={switchStyle.title} variant="label">
          {title}
        </Text>
      )}
    </View>
  );
};


```
Here is the test
```js 
  it('should render Switch with label', () => {
    render(<Switch title="activate" />);
    expect(screen.getByRole('switch', {name: 'activate'})).toBeOnTheScreen();
  });
```

## Screenshots

Here the result of the test
![image](https://github.com/callstack/react-native-testing-library/assets/47351376/d8e8c9cc-1a50-4596-834c-eaf2d7b940d7)

## Versions
react: 18.2.0
react-native: 0.73.3
react-test-renderer: 18.2.0
@testing-library/react-native: 12.4.3
